### PR TITLE
cart management and added private routes

### PR DESCRIPTION
**Pull request by praveenreddy1798**
cart

### CODE REVIEW

File: src/components/CartPrice.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      cartItems,
+      totalPrice,
+      totalDiscount,
+      totalDeliveryCharge,
+    },
+  } = useProducts();
+  return (
+    <div class="bg-white pd-md price-details-cart h-max-content">
+      <h3 class="flex-center">Price Details</h3>
+      <hr />
```
Comment (vaibhav18matere) [inline src/components/CartPrice.js:17]: I think we should not use `<hr/>` and `<br/>` tags in react. Please check once.

File: src/components/CartProduct.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  const { moveCartToWishlist } = useMoveCartToWishlist();
+  const { handleCartQuantity } = useHandleCartQuantity();
+  const wishlisted = inWishlist(wishlist, _id);
+  return (
+    <div class="card horizontal-card flex bg-white">
+      <div class="card-image-container flex-evenly">
+        <img
+          class="card-image pointer"
+          src={image}
+          alt="product"
```
Comment (vaibhav18matere) [inline src/components/CartProduct.js]: yo can give more descriptive `alt` name

**Review by vaibhav18matere — APPROVED** [review]
everything is clean, you can merge this:
(just check the following small things, confirm and change if needed)

**Review by Koushith — APPROVED** [review]
just scanned through, as @vaibhav18matere suggestions add alt-text. code structure and naming conventions LGTM. good to merge.

File: src/components/CartPrice.js (hunk 1 above)
Comment (praveenreddy1798) [reply src/components/CartPrice.js:17]: I think we can use hr tag, just br tag  we must avoid

**Pull request merged**